pkg/ghclient/core.go

```diff
@@ -46,14 +48,24 @@ type Client struct {
 }
 
 // NewClient makes a new Client with the specified token and dry-run status.
-func NewClient(token string, dryRun bool) *Client {
+func NewClient(endpoint string, token string, dryRun bool) *Client {
```

Comment:
Shouldn't we just ask the user to give us a host? We can validate that they give us a host at startup, and not make assumptions like if they give us a host and a path, etc etc etc. IMO it would be nicer to just be very clear as to what we need in the configuration value instead of these steps to default, handle bad input, etc.

Reply:
This is similar to our previous discussion. It is not convenient to derive API endpoint from GitHub host: `github.com` => `https://api.github.com` and `github.foo.com` => `https://github.foo.com/api/v3`.

I assume your concern is about this piece of code:

```
	if !strings.HasSuffix(baseURL.Path, "/") {
		baseURL.Path += "/"
	}
```

This is actually the same code in `github.NewClient(..)`.